master/lustregittagpoller.py

```diff
@@ -0,0 +1,118 @@
+# -*- python -*-
+# ex: set syntax=python:
+
+import itertools
+import os
+
+from buildbot import config
+from twisted.internet import defer
+from twisted.internet import utils
+from twisted.python import log
+from buildbot.util import epoch2datetime
+from buildbot.changes.gitpoller import GitPoller 
+
+class LustreTagPoller(GitPoller):
+
+    """This source polls a git repo for new tags and generates a change.
+
+    This source will poll a remote git repository for new tags and finds changes
+    to existing tags. Once a new or changed tag is found, a change is submitted to
+    the build master.""" 
+
+    def __init__(self, **kwargs):
+        if kwargs.get('workdir') is None:
+            kwargs['workdir'] = 'lustretagpoller-work'
+
+        GitPoller.__init__(self, **kwargs)
+
+    def describe(self):
+        str = ('LustreTagPoller: watching the remote git repository ' +
+               self.repourl + ' for new tags')
+
+        if not self.master:
+            str += " [STOPPED - check log]"
+
+        return str
+
+    def _getRefs(self):
+        """Collect and parse all remote references"""
+
+        d = self._dovccmd('show-ref', [])
+
+        @d.addCallback
+        def parseRefs(rows):
+            refs = {} 
+            for row in rows.splitlines():
+                sha, ref = row.split()
+
+                if not self._filter_ref(ref):
+                    refs[ref] = sha
+
+            return refs
+
+        return d
+
+    def _filter_ref(self, ref):
+        """Filter refs that are not tags"""
+        return not ref.startswith("refs/tags/")
+
+    @defer.inlineCallbacks
+    def _process_change(self, ref, rev):
+        """Process each tag """
```

Comment:
This comment is not terribly useful.  What does it mean to "process" a tag?  What exactly is the process that is being applied?  A tag goes in, what comes out?